src/cmds.c

```diff
@@ -853,20 +853,21 @@ static void cmd_pls_bot(struct userrec *u, int idx, char *par)
   }
 
 #ifndef TLS
-  if ((port && *port == '+') || (relay && relay[0] == '+')) {
-    dprintf(idx, "Ports prefixed with '+' are not enabled "
+  if ((port && (*port == '+' || *port == '-')) ||
+      (relay && (*relay == '+' || *relay == '-'))) {
+    dprintf(idx, "Ports prefixed with '+' or '-' are not enabled "
       "(this Eggdrop was compiled without TLS support).\n");
     return;
   }
 #endif
   if (port) {
-    if (!check_int_range(port, 0, 65536)) {
+    if (!check_int_range(port, 0, 65536) && !check_int_range(port, -65536, 0)) {
```

Comment:
maybe just check_int_range(port, -65536, 65536)?

Reply:
Still have to check for zero though.

Reply:
if (!check_int_range(abs(port), 0, 65536)) {

Reply:
Not a bad idea, on both accounts.

Reply:
i would also change that line. but go one step further. check_int_range() is always called with fixed parameters 2 and 3 within eggdrop. we dont need to keep it for compatibility, its not exportet. i propose, to rename it to something like strtoport() wich can return error if not in range and the port as an int if successful. the func only needs one param, the char*port. the rest of the code can use the returned int, and doesnt have to fight with strtol or atoi again later.

ok, not rly part of the PR, but hey, its out of my head now.

Reply:
@key2peace Good luck reusing that function to check if -9 lies between -100 and 0.
@michaelortmann I wouldn't disapprove but as you mention: not the point of this PR.

Reply:
@cizzle it is only used to check port range, the usage is the same everywhere
@michaelortmann rename it to isvalidport(port) and rip out min/max as it should be between 1 and 65535 anyways